main.go

```diff
@@ -123,9 +126,49 @@ func (lb *loadBalancer) nextProxy() *httputil.ReverseProxy {
 	return nil
 }
 
+// Returns the next backend the request should go to.
+func (lb *loadBalancer) listProxy() *httputil.ReverseProxy {
+	lb.Lock()
+	defer lb.Unlock()
+
+	tries := 0
+	listIndex := lb.listIndex
+	for {
+		var proxy *httputil.ReverseProxy
+		if lb.backends[listIndex].up {
+			proxy = lb.backends[listIndex].proxy
+		}
+		listIndex++
+		if listIndex == len(lb.backends) {
+			listIndex = 0
+		}
+		if proxy != nil {
+			return proxy
+		}
+		tries++
+		if tries == len(lb.backends) {
+			break
+		}
+	}
+	return nil
+}
+
 // ServeHTTP - LoadBalancer implements http.Handler
 func (lb *loadBalancer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	proxy := lb.nextProxy()
+	p := r.URL.Path
+	p = strings.TrimPrefix(p, "/")
+	p = strings.TrimSuffix(p, "/")
```

Comment:
Convert this into a function perhaps?

```
path2BucketObject() 
```
https://github.com/minio/minio/blob/3614cb7a8b5cb666c3e4da0f1c4993ee7d4b7db3/cmd/utils.go#L76